src/contexts/useFlashMessageContext.js

```diff
@@ -1,14 +1,16 @@
-import React, { createContext, useContext, useState } from 'react'
+import React, { createContext, useContext, useState, useCallback } from 'react'
 
 export const FlashMessageContext = createContext()
 
 export function FlashMessageProvider ({ children }) {
   const [message, setMessage] = useState('')
   const [time, setTime] = useState()
-  const newMessage = (message, time) => {
+
+  const newMessage = useCallback((message, time) => {
```

Comment:
this is the change that fixed the issue (alongside the matching change in HoloFuel). Everything else in this PR is just removing our workarounds and testing.